File: src/AgroPlatform.Domain/Audit/AuditLog.cs
using AgroPlatform.Domain.Common;

namespace AgroPlatform.Domain.Audit;

public class AuditLog : BaseEntity, ITenantEntity
{
    public Guid TenantId { get; set; }
    public string UserId { get; set; } = string.Empty;
    public string Action { get; set; } = string.Empty;
    public string EntityType { get; set; } = string.Empty;
    public string? EntityId { get; set; }
    public DateTime Timestamp { get; set; }
    public string? Metadata { get; set; }
}

File: frontend/src/types/auditLog.ts
export interface AuditLog {
  id: string;
  userId: string;
  action: string;
  entityType: string;
  entityId?: string;
  timestamp: string;
  metadata?: string;
}

export interface AuditLogsResponse {
  items: AuditLog[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}

export interface AuditLogsParams {
  userId?: string;
  dateFrom?: string;
  dateTo?: string;
  entityType?: string;
  action?: string;
  page?: number;
  pageSize?: number;
}
